rvgo/test/syscall_test.go

```diff
@@ -0,0 +1,903 @@
+			var syscallErr *fast.UnsupportedSyscallErr
+			require.ErrorAs(t, err, &syscallErr)
+
+			// TODO: Test EVM & slow VM
+		})
+	}
+}
+
+func FuzzStateSyscallExit(f *testing.F) {
+	contracts := testContracts(f)
+	addrs := testAddrs
+
+	syscalls := []int{fast.SysExit, fast.SysExitGroup}
+
+	testExit := func(t *testing.T, syscall int, exitCode uint8, pc uint64, step uint64) {
+		pc = pc & 0xFF_FF_FF_FF_FF_FF_FF_FC // align PC
+		state := &fast.VMState{
+			PC:              pc,
+			Heap:            0,
+			ExitCode:        0,
+			Exited:          false,
+			Memory:          fast.NewMemory(),
+			LoadReservation: 0,
+			Registers:       [32]uint64{17: uint64(syscall), 10: uint64(exitCode)},
+			Step:            step,
+		}
+		state.Memory.SetUnaligned(pc, syscallInsn)
+		preStateRoot := state.Memory.MerkleRoot()
+		preStateRegisters := state.Registers
+
+		fastState := fast.NewInstrumentedState(state, nil, os.Stdout, os.Stderr)
+		stepWitness, err := fastState.Step(true)
+		require.NoError(t, err)
+		require.False(t, stepWitness.HasPreimage())
+
+		require.Equal(t, pc+4, state.PC) // PC must advance
+		require.Equal(t, uint64(0), state.Heap)
+		require.Equal(t, uint64(0), state.LoadReservation)
+		require.Equal(t, exitCode, state.ExitCode) // ExitCode must be set
+		require.Equal(t, true, state.Exited)       // Must be exited
+		require.Equal(t, preStateRoot, state.Memory.MerkleRoot())
+		require.Equal(t, step+1, state.Step) // Step must advance
+		require.Equal(t, preStateRegisters, state.Registers)
+
+		fastPost := state.EncodeWitness()
+		runEVM(t, contracts, addrs, stepWitness, fastPost)
+		runSlow(t, stepWitness, fastPost, nil)
+	}
+
+	f.Fuzz(func(t *testing.T, exitCode uint8, pc uint64, step uint64) {
+		for _, syscall := range syscalls {
+			testExit(t, syscall, exitCode, pc, step)
+		}
+	})
+}
+
+func FuzzStateSyscallBrk(f *testing.F) {
+	contracts := testContracts(f)
+	addrs := testAddrs
+
+	f.Fuzz(func(t *testing.T, pc, step uint64) {
+		pc = pc & 0xFF_FF_FF_FF_FF_FF_FF_FC // align PC
+		state := &fast.VMState{
+			PC:              pc,
+			Heap:            0,
+			ExitCode:        0,
+			Exited:          false,
+			Memory:          fast.NewMemory(),
+			LoadReservation: 0,
+			Registers:       [32]uint64{17: fast.SysBrk},
+			Step:            step,
+		}
+		state.Memory.SetUnaligned(pc, syscallInsn)
+		preStateRoot := state.Memory.MerkleRoot()
+		expectedRegisters := state.Registers
+		expectedRegisters[10] = 1 << 30
+		expectedRegisters[11] = 0
+
+		fastState := fast.NewInstrumentedState(state, nil, os.Stdout, os.Stderr)
+		stepWitness, err := fastState.Step(true)
+		require.NoError(t, err)
+		require.False(t, stepWitness.HasPreimage())
+
+		require.Equal(t, pc+4, state.PC) // PC must advance
+		require.Equal(t, uint64(0), state.Heap)
+		require.Equal(t, uint64(0), state.LoadReservation)
+		require.Equal(t, uint8(0), state.ExitCode) // ExitCode must be set
+		require.Equal(t, false, state.Exited)      // Must be exited
```

Comment:
```suggestion
		require.Equal(t, false, state.Exited)      // Must not be exited
```
also other syscalls' comment must be fixed.